internal/sbi/producer/ue_context.go

```diff
@@ -257,11 +261,47 @@ func UEContextTransferProcedure(ueContextID string, ueContextTransferRequest mod
 	switch UeContextTransferReqData.Reason {
 	case models.TransferReason_INIT_REG:
 		// TODO: check integrity of the registration request included in ueContextTransferRequest
+		_, integrityProtected, err := nas_security.Decode(ue, UeContextTransferReqData.AccessType,
+			ueContextTransferRequest.BinaryDataN1Message, true)
+		if err != nil {
+			ue.NASLog.Errorln(err)
+		}
+		// ue.MacFailed = !integrityProtected
+		if integrityProtected {
+			ueContextTransferRspData.UeContext = buildUEContextModel(ue, integrityProtected)
+		} else {
+			problemDetails := &models.ProblemDetails{
+				Status: http.StatusForbidden,
+				Cause:  "INTEGRITY_CHECK_FAIL",
+				InvalidParams: []models.InvalidParam{
+					{
+						Param: "reason",
+					},
+				},
+			}
+			return nil, problemDetails
+		}
 		// TODO: handle condition of TS 29.518 5.2.2.2.1.1 step 2a case b
-		ueContextTransferRspData.UeContext = buildUEContextModel(ue)
 	case models.TransferReason_MOBI_REG:
-		// TODO: check integrity of the registration request included in ueContextTransferRequest
-		ueContextTransferRspData.UeContext = buildUEContextModel(ue)
+		_, integrityProtected, err := nas_security.Decode(ue, UeContextTransferReqData.AccessType,
+			ueContextTransferRequest.BinaryDataN1Message, false)
+		if err != nil {
+			ue.NASLog.Errorln(err)
+		}
+		if integrityProtected {
+			ueContextTransferRspData.UeContext = buildUEContextModel(ue, integrityProtected)
+		} else {
+			problemDetails := &models.ProblemDetails{
+				Status: http.StatusForbidden,
+				Cause:  "INTEGRITY_CHECK_FAIL",
+				InvalidParams: []models.InvalidParam{
+					{
+						Param: "reason",
+					},
+				},
+			}
+			return nil, problemDetails
+		}
```

Comment:
Please make this part a common api since it duplicates with that of `case models.TransferReason_MOBI_REG_UE_VALIDATED`.